packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -177,6 +177,64 @@ export function MediaPlaceholder( {
 		} );
 	};
 
+	async function onHTMLDrop( HTML ) {
```

Comment:
@ntsekouras @youknowriad What do you think of this approach? There are still something missing: handling of other media types, success/error notices (?), error handling, but the general idea is in place. Any concerns you see?

Reply:
I like the approach of the MediaPlaceholder taking handling this instead of blocks. It makes sense but why handle just the HTML case, why not handle the generic `wp-blocks` case? (If we're drag and dropping and image block or a video block... handle it in the same way you're doing here).

For me `onHTMLDrop` is less about block HTML being drag and dropped but more about random HTML being dropped (you can try by drag and dropping random HTML from random sites). So in other words, if we were to define `onHTMLDrop` in MediaPlaceholder, I expect it to be able to handle the case where I drag an image from an external website.

Reply:
> This callback handles that specific case. What am I missing?

You're right, I mis read the code.

That said, I still think wp-blocks and random HTML should be treated differently similarly to how we do it in `useBlockDropZone`

Basically: 

 - `wp-blocks` => use `wp.blocks.parse` to parse the blocks (the content of the event is already a list of blocks, there's not need to run paste handler to transform random HTML to blocks)
- Random HTML: Random HTML is being dragged from a random third-party site (not using blocks), we use the pasteHandle to guess to a maximum what kind of content is being dragged.

Reply:
Updated!